lib/computer.rb

```diff
@@ -1,15 +1,50 @@
-require "./spec/spec_helper"
-
 class Computer
-  attr_reader :mark
+  attr_reader :mark, :opponent
 
-  def initialize(mark)
-    @moves = computer_player
+  def initialize(mark, board)
     @mark = mark
+    @board = board
   end
 
   def get_move
-    @moves.shift
-    # return a random available move
+    analyse_board
```

Comment:
I see why you extracted a method here, but you've just moved the problem of "long method" into another method. What I want to see in this method are the high-level ideas for how you figure out a move. Should read like the rules that you're implementing. I'd inline analyse_board and extract more than one method from all of that code. Group lines of code together and give them a cohesive name.

Reply:
resolved #32